### ISSUE TITLE
SlackLogger respects configuration of other loggers

### ISSUE THREAD

**Issue by severisv** [severisv/SlackLogger#29]
Add this to appsettings prevents SlackLogger from posting log messages:
![image](https://user-images.githubusercontent.com/966063/184826150-f3148931-1a42-4c07-9555-e86567c42402.png)

It should only respect the general or SlackLogger-specific configuration

**Comment by pzaj2** [comment]
@severisv Any timelines on this? 

I believe your `Slack` key may be incorrect and it should be `LoggerProvider` (unfortunately, due to its name) unless `Microsoft.Extensions.Logging.ProviderAliasAttribute("Slack")` is used on the provider.

_Disclaimer_: I haven't tested it, but that's how it seems to work in general. So the issue may not be that it respects configuration from other loggers but instead, it takes the default (don't know if that makes sense because the screenshot does not show the entire configuration).

I wouldn't want to claim that any of the above is correct, however, it's a possibility worth exploring.

**Comment by pzaj2** [comment]
Upon further investigation, I believe the bug is located here: https://github.com/severisv/SlackLogger/blob/c46b8f59e70f303afc748d16ed048b2c10ab1961/SlackLogger/Settings.cs#L62

Imo, it should first look for a more specific (i.e. provider-specific) rule before applying the global rule (one where `ProviderName` is null/empty).

I believe something like this would suffice:
```
Index: SlackLogger/LoggerProvider.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/SlackLogger/LoggerProvider.cs b/SlackLogger/LoggerProvider.cs
--- a/SlackLogger/LoggerProvider.cs	(revision c46b8f59e70f303afc748d16ed048b2c10ab1961)
+++ b/SlackLogger/LoggerProvider.cs	(date 1661257541673)
@@ -6,9 +6,10 @@
 
 namespace SlackLogger
 {
+    [ProviderAlias(ProviderAlias)]
     public class LoggerProvider : ILoggerProvider
     {
-
+        public const string ProviderAlias = "Slack";
         private readonly SlackLoggerOptions _options;
         private readonly ScopeSettings _scopeSettings;
 
Index: SlackLogger/Settings.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/SlackLogger/Settings.cs b/SlackLogger/Settings.cs
--- a/SlackLogger/Settings.cs	(revision c46b8f59e70f303afc748d16ed048b2c10ab1961)
+++ b/SlackLogger/Settings.cs	(date 1661257773435)
@@ -58,8 +58,11 @@
                 level = LogLevel.None;
                 return false;
             }
-
-            var value = switches.FirstOrDefault(s => s.CategoryName == name)?.LogLevel;
+            
+            var value = 
+                switches.FirstOrDefault(x => x.ProviderName == LoggerProvider.ProviderAlias && x.CategoryName == name)?.LogLevel ??
+                switches.FirstOrDefault(x => x.ProviderName == LoggerProvider.ProviderAlias && x.CategoryName == null)?.LogLevel ??
+                switches.FirstOrDefault(s => s.CategoryName == name && string.IsNullOrEmpty(s.ProviderName))?.LogLevel;
             if (value == null)
             {
                 level = LogLevel.None;

```

**Comment by severisv** [comment]
Thanks for looking into it! I'm afraid I can't give a timeline, but I'm happy to handle pull requests pretty quickly if someone else wants to have a look ☺

**Comment by severisv** [comment]
Fixed by #32
